### ISSUE TITLE
Dose the trt model can deploy with triton server？

### ISSUE THREAD

**Issue by jiejie1993** [open-mmlab/mmdetection#5361]
I convert the Yolov3 model torch->onnx->tensorrt, but cannot deploy in triton server docker image. How can the model deploy with triton server?

**Comment by hhaAndroid** [comment]
Hi @jiejie1993 We haven't verified it above, but it should be possible. Can you post the error message?

**Comment by jiejie1993** [comment]
I convert the YoloV3 throw the command:`python tools/deployment/onnx2tensorrt.py checkpoints/epoch_101_backend.onnx --trt-file checkpoints/epoch_101_backend.trt --input-img tests/data/Angelas_1.jpg --shape 352 608 --mean 0 0 0 --std 255 255 255 --show --verify`

then I organize the trt model like that:`triton_repo
    ├── YoloV3
    │   └── 1
    │       └── epoch_101_backend.plan  # (rename from epoch_101_backend.trt)'

I start the docker `docker run --runtime=nvidia --network=host -it --rm -v /home/gyj/1.warp_work/0.project/2.Auto_cv/mmdetection/mmdetection/checkpoints/triton_repo:/repo  nvcr.io/nvidia/tritonserver:20.08-py3 bash`
and boot the triton server:`/opt/tritonserver/bin/tritonserver --model-store=/repo/ --strict-model-config=false --log-verbose 1`

The failed log is:
`root@gyj-MS-7B23:/opt/tritonserver# /opt/tritonserver/bin/tritonserver --model-store=/repo/ --strict-model-config=false --log-verbose 1
I0615 06:36:25.415024 60 metrics.cc:184] found 1 GPUs supporting NVML metrics
I0615 06:36:25.420698 60 metrics.cc:193]   GPU 0: GeForce RTX 2060
I0615 06:36:25.421121 60 server.cc:119] Initializing Triton Inference Server
I0615 06:36:25.551373 60 pinned_memory_manager.cc:195] Pinned memory pool is created at '0x7fb32a000000' with size 268435456
I0615 06:36:25.552140 60 netdef_backend_factory.cc:46] Create NetDefBackendFactory
I0615 06:36:25.552151 60 plan_backend_factory.cc:48] Create PlanBackendFactory
I0615 06:36:25.552156 60 plan_backend_factory.cc:55] Registering TensorRT Plugins
I0615 06:36:25.552183 60 logging.cc:52] Registered plugin creator - ::BatchTilePlugin_TRT version 1
I0615 06:36:25.552192 60 logging.cc:52] Registered plugin creator - ::BatchedNMS_TRT version 1
I0615 06:36:25.552197 60 logging.cc:52] Registered plugin creator - ::CoordConvAC version 1
I0615 06:36:25.552202 60 logging.cc:52] Registered plugin creator - ::CropAndResize version 1
I0615 06:36:25.552207 60 logging.cc:52] Registered plugin creator - ::DetectionLayer_TRT version 1
I0615 06:36:25.552212 60 logging.cc:52] Registered plugin creator - ::FlattenConcat_TRT version 1
I0615 06:36:25.552218 60 logging.cc:52] Registered plugin creator - ::GenerateDetection_TRT version 1
I0615 06:36:25.552225 60 logging.cc:52] Registered plugin creator - ::GridAnchor_TRT version 1
I0615 06:36:25.552232 60 logging.cc:52] Registered plugin creator - ::GridAnchorRect_TRT version 1
I0615 06:36:25.552240 60 logging.cc:52] Registered plugin creator - ::InstanceNormalization_TRT version 1
I0615 06:36:25.552246 60 logging.cc:52] Registered plugin creator - ::LReLU_TRT version 1
I0615 06:36:25.552251 60 logging.cc:52] Registered plugin creator - ::MultilevelCropAndResize_TRT version 1
I0615 06:36:25.552282 60 logging.cc:52] Registered plugin creator - ::MultilevelProposeROI_TRT version 1
I0615 06:36:25.552291 60 logging.cc:52] Registered plugin creator - ::NMS_TRT version 1
I0615 06:36:25.552297 60 logging.cc:52] Registered plugin creator - ::Normalize_TRT version 1
I0615 06:36:25.552309 60 logging.cc:52] Registered plugin creator - ::PriorBox_TRT version 1
I0615 06:36:25.552316 60 logging.cc:52] Registered plugin creator - ::ProposalLayer_TRT version 1
I0615 06:36:25.552323 60 logging.cc:52] Registered plugin creator - ::Proposal version 1
I0615 06:36:25.552329 60 logging.cc:52] Registered plugin creator - ::PyramidROIAlign_TRT version 1
I0615 06:36:25.552335 60 logging.cc:52] Registered plugin creator - ::Region_TRT version 1
I0615 06:36:25.552341 60 logging.cc:52] Registered plugin creator - ::Reorg_TRT version 1
I0615 06:36:25.552347 60 logging.cc:52] Registered plugin creator - ::ResizeNearest_TRT version 1
I0615 06:36:25.552354 60 logging.cc:52] Registered plugin creator - ::RPROI_TRT version 1
I0615 06:36:25.552360 60 logging.cc:52] Registered plugin creator - ::SpecialSlice_TRT version 1
I0615 06:36:25.552367 60 onnx_backend_factory.cc:53] Create OnnxBackendFactory
I0615 06:36:25.556905 60 libtorch_backend_factory.cc:53] Create LibTorchBackendFactory
I0615 06:36:25.556919 60 custom_backend_factory.cc:46] Create CustomBackendFactory
I0615 06:36:25.556924 60 backend_factory.h:44] Create TritonBackendFactory
I0615 06:36:25.556934 60 ensemble_backend_factory.cc:47] Create EnsembleBackendFactory
I0615 06:36:25.557104 60 autofill.cc:142] TensorFlow SavedModel autofill: Internal: unable to autofill for 'YoloV3', unable to find savedmodel directory named 'model.savedmodel'
I0615 06:36:25.557128 60 autofill.cc:155] TensorFlow GraphDef autofill: Internal: unable to autofill for 'YoloV3', unable to find graphdef file named 'model.graphdef'
I0615 06:36:25.557175 60 autofill.cc:168] PyTorch autofill: Internal: unable to autofill for 'YoloV3', unable to find PyTorch file named 'model.pt'
I0615 06:36:25.557208 60 autofill.cc:180] Caffe2 NetDef autofill: Internal: unable to autofill for 'YoloV3', unable to find netdef files: 'model.netdef' and 'init_model.netdef'
WARNING: Since openmp is enabled in this build, this API cannot be used to configure intra op num threads. Please use the openmp environment variables to control the number of threads.
I0615 06:36:25.902978 60 autofill.cc:376] failed to load /repo/YoloV3/1/epoch_101_backend.plan: Internal: onnx runtime error 1: /workspace/onnxruntime/onnxruntime/core/session/inference_session.cc:279 onnxruntime::InferenceSession::InferenceSession(const onnxruntime::SessionOptions&, const onnxruntime::Environment&, const void*, int) result was false. Could not parse model successfully while constructing the inference session

I0615 06:36:25.903043 60 autofill.cc:212] ONNX autofill: Internal: unable to autofill for 'YoloV3', unable to find onnx file
E0615 06:36:38.047974 60 logging.cc:43] coreReadArchive.cpp (38) - Serialization Error in verifyHeader: 0 (Version tag does not match)
E0615 06:36:38.051499 60 logging.cc:43] INVALID_STATE: std::exception
E0615 06:36:38.051512 60 logging.cc:43] INVALID_CONFIG: Deserialize the cuda engine failed.
I0615 06:36:38.071110 60 autofill.cc:225] TensorRT autofill: Internal: unable to autofill for 'YoloV3', unable to find a compatible plan file.
W0615 06:36:38.071140 60 autofill.cc:265] Proceeding with simple config for now
I0615 06:36:38.071146 60 model_config_utils.cc:629] autofilled config: name: "YoloV3"

E0615 06:36:38.071540 60 model_repository_manager.cc:1633] unexpected platform type  for YoloV3
error: creating server: Internal - failed to load all models
`

**Comment by hhaAndroid** [comment]
Hi @jiejie1993 You can use onnxruntime for inference first without docker, it is easy to judge the cause of the error at this time.

**Comment by jiejie1993** [comment]
I have check the onnx model with onnxruntime, it works fine.
`python tools/deployment/test.py configs/yolo/yolov3_d53_mstrain-608_273e_voc_haiyangshengwu.py checkpoints/epoch_101_backend.onnx --out log_model_dir/haiyangshengwu_YoloV3/out_result.pkl --backend onnxruntime --eval mAP --show-dir log_model_dir/haiyangshengwu_YoloV3/`

**Comment by hhaAndroid** [comment]
> I have check the onnx model with onnxruntime, it works fine.
> `python tools/deployment/test.py configs/yolo/yolov3_d53_mstrain-608_273e_voc_haiyangshengwu.py checkpoints/epoch_101_backend.onnx --out log_model_dir/haiyangshengwu_YoloV3/out_result.pkl --backend onnxruntime --eval mAP --show-dir log_model_dir/haiyangshengwu_YoloV3/`
> 
> the val result
> `
> [>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 240/240, 3.0 task/s, elapsed: 80s, ETA: 0s
> writing results to log_model_dir/haiyangshengwu_YoloV3/out_result.pkl
> 
> ---------------iou_thr: 0.5---------------
> 
> +--------------------+-----+------+--------+-------+
> | class | gts | dets | recall | ap |
> +--------------------+-----+------+--------+-------+
> | angelas | 230 | 8419 | 0.922 | 0.791 |
> | angel_fish | 305 | 5567 | 0.951 | 0.925 |
> | Apple_Mermaid | 122 | 5010 | 0.975 | 0.929 |
> | Ball_photoelectric | 127 | 5004 | 0.953 | 0.921 |
> +--------------------+-----+------+--------+-------+
> | mAP | | | | 0.892 |
> +--------------------+-----+------+--------+-------+
> OrderedDict([('AP50', 0.892), ('mAP', 0.8916489481925964)])
> `

This mean that you need to check the docker settings.
